python/lsst/pipe/tasks/calibrate.py

```diff
@@ -444,6 +547,39 @@ def runDataRef(self, dataRef, exposure=None, background=None, icSourceCat=None,
 
         return calRes
 
+    def adaptArgsAndRun(self, inputData, inputDataIds, outputDataIds, butler):
+        expId, expBits = butler.registry.packDataId("VisitDetector",
+                                                    inputDataIds['exposure'],
+                                                    returnMaxBits=True)
+        inputData['exposureIdInfo'] = ExposureIdInfo(expId, expBits)
+
+        if self.config.doAstrometry:
+            refObjLoader = ReferenceObjectLoader(inputDataIds['astromRefCat'],
+                                                 butler,
+                                                 self.config.astromRefObjLoader,
+                                                 self.log)
+            self.pixelMargin = refObjLoader.config.pixelMargin
+            self.astrometry.setRefObjLoader(refObjLoader)
+
+        if self.config.doPhotoCal:
+            photoRefObjLoader = ReferenceObjectLoader(inputDataIds['photoRefCat'],
+                                                      butler,
+                                                      self.config.photoRefObjLoader,
+                                                      self.log)
+            self.pixelMargin = photoRefObjLoader.config.pixelMargin
+            self.photoCal.match.setRefObjLoader(photoRefObjLoader)
+
+        results = self.run(**inputData)
+
+        if self.config.doWriteMatches and results.astromMatches is not None:
```

Comment:
Is it possible for the first clause here to be true while the second is false?  If so, that could be a problem in the future (because we're lying to the activator about what it should expect to see in the outputs).  If we can easily convince ourselves that `if self.config.doWriteMatches` alone is sufficient here now, that'd be ideal; if not, I'd like to at least add a warning if `results.astromMatrches` is unexpectedly `None` and open a ticket to follow up later.